src/goalServices/goals.js

```diff
@@ -1277,7 +1305,34 @@ export async function saveGoalsForReport(goals, report) {
     },
   });
 
-  const currentGoals = await Promise.all(goals.map(async (goal) => {
+  // Loop and Create or Update goals.
+  const currentGoals = await Promise.all(goals.map(async (goal, index) => {
+    // We need to skip creation of monitoring goals for non monitoring grants.
+    if (goal.standard === 'Monitoring') {
+      // Find the corresponding monitoring goals.
+      const monitoringGoals = await Goal.findAll({
+        attribute: ['grantId'],
+        raw: true,
+        where: {
+          grantId: goal.grantIds,
+          createdVia: 'monitoring',
+          status: { [Op.not]: GOAL_STATUS.CLOSED },
+        },
+      });
+      if (monitoringGoals.length > 0) {
+        // Replace the goal granIds only with the grants that should have monitoring goals created.
+        // eslint-disable-next-line no-param-reassign
+        goals[index].grantIds = monitoringGoals;
```

Comment:
this evaluates to goals[index].grantIds = [{ grantId: 1}, {grantId: 2}] so I don't think this is right

Reply:
Yea im debugging this atm will fix.

Reply:
The idea here was that raw: true would return a array of ids.